include/laser_filters/polygon_filter.h

```diff
@@ -0,0 +1,608 @@
+    auto update_elapsed = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+
+    RCLCPP_DEBUG(logging_interface_->get_logger(), "LaserScanPolygonFilter update took %lu microseconds", update_elapsed);
+
+    return true;
+  }
+
+private:
+  laser_geometry::LaserProjection projector_;
+};
+
+/**
+ * @brief This is a filter that removes points in a laser scan inside of a polygon.
+ * It assumes that the transform between the scanner and the robot base remains unchanged,
+ * i.e. the position and orientation of the laser filter should not change.
+ * A typical use case for this filter is to filter out parts of the robot body or load that it may carry.
+ */
+class StaticLaserScanPolygonFilter : public LaserScanPolygonFilterBase {
+public:
+  bool configure() override
+  {
+    bool result = LaserScanPolygonFilterBase::configure();
+    is_polygon_transformed_ = false;
+    transform_timeout_ = 5; // Default
+    if (!filters::FilterBase<sensor_msgs::msg::LaserScan>::getParam(std::string("transform_timeout"), transform_timeout_))
+    {
+      RCLCPP_INFO(logging_interface_->get_logger(), "Error: PolygonFilter transform_timeout not set, assuming 5. \n");
+    }
+    return result;
+  }
+
+  bool update(const sensor_msgs::msg::LaserScan& input_scan, sensor_msgs::msg::LaserScan& output_scan) override
+  {
+    boost::recursive_mutex::scoped_lock lock(own_mutex_);
+    publishPolygon();
+
+    if (!is_polygon_transformed_) 
+    {
+      if (!transformPolygon(input_scan.header.frame_id)) return false;
+    }
+
+    output_scan = input_scan;
+    checkCoSineMap(input_scan);
+
+    size_t i = 0;
+    size_t i_max = input_scan.ranges.size();
+
+    while (i < i_max)
+    {
+      float range = input_scan.ranges[i];
+
+      float x = co_sine_map_(i, 0) * range;
+      float y = co_sine_map_(i, 1) * range;
+      Point point(x, y, 0);
+
+      if (invert_filter_ != inPolygon(point))
+      {
+        output_scan.ranges[i] = std::numeric_limits<float>::quiet_NaN();
+      }
+
+      ++i;
+    }
+
+    return true;
+    }
+
+  void footprintCB(const geometry_msgs::msg::Polygon::SharedPtr polygon)
+  {
+    is_polygon_transformed_ = false;
+    LaserScanPolygonFilterBase::footprintCB(polygon);
+  }
+  
+protected:
+  bool transformPolygon(const std::string &input_scan_frame_id)
+  {
+    std::string error_msg;
+    RCLCPP_DEBUG(logging_interface_->get_logger(),
+      "waitForTransform %s -> %s",
+      polygon_frame_.c_str(), input_scan_frame_id.c_str()
+    );
+    
+    geometry_msgs::msg::TransformStamped transform;
+    try
+    {
+      transform = buffer_.lookupTransform(input_scan_frame_id,
+        polygon_frame_,
+        tf2::TimePointZero,
+        tf2::durationFromSec(transform_timeout_));
+    }
+    catch(tf2::TransformException& ex)
+    {
+      RCLCPP_WARN_THROTTLE(logging_interface_->get_logger(),
+          *get_clock(), 1000,
+          "StaticLaserScanPolygonFilter",
```

Comment:
`warning: too many arguments for format`. This seems like a leftover from converting `ROS_WARN_THROTTLE_NAMED`

```suggestion
```